src/pycolmap/estimators/covariance.cc

```diff
@@ -0,0 +1,92 @@
+#include "colmap/estimators/covariance.h"
+
+#include "pycolmap/pybind11_extension.h"
+
+#include <pybind11/eigen.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/stl_bind.h>
+
+using namespace colmap;
+using namespace pybind11::literals;
+namespace py = pybind11;
+
+void BindCovarianceEstimator(py::module& m) {
+  m.def(
+      "estimate_pose_covariance_from_ba_ceres_backend",
+      [](ceres::Problem* problem,
+         Reconstruction* reconstruction) -> py::object {
+        std::map<image_t, Eigen::MatrixXd> res;
+        if (!EstimatePoseCovarianceCeresBackend(problem, reconstruction, res))
+          return py::none();
+        return py::cast(res);
+      },
+      py::arg("problem"),
+      py::arg("reconstruction"));
+
+  m.def(
+      "estimate_pose_covariance_from_ba",
+      [](ceres::Problem* problem,
+         Reconstruction* reconstruction,
+         double lambda) -> py::object {
+        std::map<image_t, Eigen::MatrixXd> res;
+        if (!EstimatePoseCovariance(problem, reconstruction, res, lambda))
+          return py::none();
+        return py::cast(res);
+      },
+      py::arg("problem"),
+      py::arg("reconstruction"),
+      py::arg("lambda") = 1e-6);
+
+#define REGISTER_ESTIMATOR(NAME)                                               \
+  .def(py::init<ceres::Problem*, Reconstruction*>())                           \
+      .def("compute", &NAME::Compute)                                          \
+      .def("compute_full", &NAME::ComputeFull)                                 \
+      .def("get_pose_covariance",                                              \
+           (Eigen::MatrixXd(NAME::*)() const) & NAME::GetPoseCovariance)       \
```

Comment:
Use `py::overload_cast` instead of C-style cast here and below: https://pybind11.readthedocs.io/en/stable/classes.html#overloaded-methods